simphony/io/h5_cuds.py

```diff
@@ -152,6 +155,47 @@ def add_mesh(self, mesh):
         self._file.flush()
         return m
 
+    def add_lattice(self, lattice, record=None):
```

Comment:
"Is this related to a CUBA keywords or is part of the internal representation of a specific engine? Do all engines will need to save these values? If the float vector was part of the CUBA keywords that are saved wouldn't that remove the need for extended the table columns?"

I think we are discussing of slightly different topics here. We do not (often) have to extend a FileLattice table after its creation. Typically before the simulation, we already know what quantities are relevant, and a FileLattice can be created with empty columns for those quantities. The problem is that a Lattice can have something else stored in it (like this "nearly incompressible" 19-dim vector array used in a certain Lattice-Boltzmann model).

We can move this discussion elsewhere and I can modify the PR so that partial saving is only supported with FileLattice constructor for now.